sandbox/lua/decoders/bind_query_log.lua

```diff
@@ -0,0 +1,274 @@
+
+The ', "record_type"' part sets the name of the capture's entry in the table of
+matches that gets built.
+
+Source: https://en.wikipedia.org/wiki/List_of_DNS_record_types
+--]]
+dns_record_type = l.Cg(
+      l.P"A" /"A"
+    + l.P"AAAA" /"AAAA"
+    + l.P"AFSDB" /"AFSDB"
+    + l.P"APL" /"APL"
+    + l.P"AXFR" /"AXFR"
+    + l.P"CAA" /"CAA"
+    + l.P"CDNSKEY" /"CDNSKEY"
+    + l.P"CDS" /"CDS"
+    + l.P"CERT" /"CERT"
+    + l.P"CNAME" /"CNAME"
+    + l.P"DHCID" /"DHCID"
+    + l.P"DLV" /"DLV"
+    + l.P"DNAME" /"DNAME"
+    + l.P"DS" /"DS"
+    + l.P"HIP" /"HIP"
+    + l.P"IPSECKEY" /"IPSECKEY"
+    + l.P"IXFR" /"IXFR"
+    + l.P"KEY" /"KEY"
+    + l.P"KX" /"KX"
+    + l.P"LOC" /"LOC"
+    + l.P"MX" /"MX"
+    + l.P"NAPTR" /"NAPTR"
+    + l.P"NS" /"NS"
+    + l.P"NSEC" /"NSEC"
+    + l.P"NSEC3" /"NSEC3"
+    + l.P"NSEC3PARAM" /"NSEC3PARAM"
+    + l.P"OPT" /"OPT"
+    + l.P"PTR" /"PTR"
+    + l.P"RRSIG" /"RRSIG"
+    + l.P"RP" /"RP"
+    + l.P"SIG" /"SIG"
+    + l.P"SOA" /"SOA"
+    + l.P"SRV" /"SRV"
+    + l.P"SSHFP" /"SSHFP"
+    + l.P"TA" /"TA"
+    + l.P"TKEY" /"TKEY"
+    + l.P"TLSA" /"TLSA"
+    + l.P"TSIG" /"TSIG"
+    + l.P"TXT" /"TXT"
+    + l.P"*" /"*"
+    , "RecordType")
+
+--A capture group for the 3 kinds of DNS record classes.
+-- Source: https://en.wikipedia.org/wiki/Domain_Name_System#DNS_resource_records
+dns_record_class = l.Cg(
+      --For internet records:
+      l.P"IN" /"IN"
+      --For CHAOS records:
+    + l.P"CH" /"CH"
+      --For Hesiod records:
+    + l.P"HS" /"HS"
+    , "RecordClass")
+
+--[[Query flag patterns
+Sources on what the query flags mean: 
+
+* https://deepthought.isc.org/article/AA-00434/0/What-do-EDC-and-other-letters-I-see-in-my-query-log-mean.html
+* http://jpmens.net/2011/02/22/bind-querylog-know-your-flags/
+
+]]--
+
+--Create a table to hold possible matches that may appear:
+local t = {
+      ["+"] = "recursion requested",
+      ["-"] = "recursion not requested",
+      E = "EDNS used",
+      S = "query signed",
+      D = "DNSSEC data wanted",
+      C = "no DNSSEC validation check",
+      T = "TCP used",
+}
+
+--Create a capture group that uses the table above to match any one or more of '+-EDCST'
+--that are present and add them to a new table called QueryFlags: 
+query_flags = l.Cg( l.Ct((l.S"+-EDCST" / t)^1), "QueryFlags" )
+
+--[[Hostname and domain name patterns
+
+Hostnames and domain names are broken up into fragments that are called 
+"labels", which are the parts between the dots.
+
+Source: https://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_host_names
+
+For instance, webserver.company.com has the labels "webserver", "company" and 
+"com"
+
+The pattern below uses the upper, lower and digit shortcuts from the main LPEG 
+library and combines them with the hyphen (-) character to match anything that's
+part of a valid hostname label. The ^1 means match one or more instances of it.
+--]]
+local hostname_fragment = (l.upper + l.lower + l.digit +  "-")^1
```

Comment:
alnum would suffice for the first three but a common hostname grammar should be used instead (there is an open issue since we are getting quite the collection of hostname parsers) https://github.com/mozilla-services/lua_sandbox/issues/129